api/src/test/java/marquez/db/DbTestUtils.java

```diff
@@ -0,0 +1,243 @@
+  static RunArgsRow newRunArgs(final Jdbi jdbi) {
+    final RunArgsDao runArgsDao = jdbi.onDemand(RunArgsDao.class);
+    final ImmutableMap<String, String> runArgs = marquez.service.models.ModelGenerator.newRunArgs();
+    final String runArgsAsJson = Utils.toJson(runArgs);
+    final String checksum = Utils.checksumFor(runArgs);
+    return runArgsDao.upsertRunArgs(newRowUuid(), newTimestamp(), runArgsAsJson, checksum);
+  }
+
+  /** Adds a new {@link RunRow} object to the {@code runs} table. */
+  static RunRow newRun(final Jdbi jdbi, final String namespaceName, final String jobName) {
+    final RunDao runDao = jdbi.onDemand(RunDao.class);
+    final RunMeta runMeta = newRunMeta();
+    return runDao.upsertRunMeta(
+        NamespaceName.of(namespaceName), JobName.of(jobName), runMeta, RunState.NEW);
+  }
+
+  /** Adds a new {@link RunRow} object to the {@code runs} table. */
+  static ExtendedRunRow newRun(
+      final Jdbi jdbi,
+      final UUID jobVersionUuid,
+      final UUID runArgsUuid,
+      final UUID namespaceUuid,
+      final String namespaceName,
+      final String jobName,
+      final String jobLocation,
+      final UUID jobContextUuid) {
+    final RunDao runDao = jdbi.onDemand(RunDao.class);
+    return runDao.upsert(
+        newRowUuid(),
+        newExternalId(),
+        newTimestamp(),
+        jobVersionUuid,
+        runArgsUuid,
+        newTimestamp(),
+        newTimestamp(),
+        namespaceUuid,
+        namespaceName,
+        jobName,
+        jobLocation,
+        jobContextUuid);
+  }
+
+  /** Transition a {@link Run} to the provided {@link RunState}. */
+  static Run transitionRunTo(final Jdbi jdbi, final UUID runUuid, final RunState runState) {
+    final RunDao runDao = jdbi.onDemand(RunDao.class);
+    final RunStateDao runStateDao = jdbi.onDemand(RunStateDao.class);
+    runStateDao.updateRunStateFor(runUuid, runState, newTimestamp());
+    return runDao.findRunByUuid(runUuid).get();
+  }
+
+  /**
+   * Transition a {@link Run} to the provided {@link RunState} and associates the specified inputs
+   * to the run.
+   */
+  static Run transitionRunWithOutputs(
+      final Jdbi jdbi,
+      final UUID runUuid,
+      final RunState runState,
+      final ImmutableSet<DatasetId> runOutputIds) {
+    final Run run = transitionRunTo(jdbi, runUuid, runState);
+    final RunDao runDao = jdbi.onDemand(RunDao.class);
+    runDao.upsertOutputDatasetsFor(runUuid, runOutputIds);
+    return run;
+  }
+
+  /** Verifies a {@link Run} has {@code expected} input dataset versions. */
+  static void verifyRunHasInputs(final Jdbi jdbi, final UUID runUuid, final int expected) {
+    final DatasetVersionDao datasetVersionDao = jdbi.onDemand(DatasetVersionDao.class);
+    final List<ExtendedDatasetVersionRow> inputDatasetVersions =
+        datasetVersionDao.findInputDatasetVersionsFor(runUuid);
+    assertThat(inputDatasetVersions).hasSize(expected);
+  }
```

Comment:
My personal preference is to always keep my assertions in the test method. Especially when the logic is only a couple of lines- the code repetition is a small price to pay in order to ensure all tests are very clear in their assertions.